internal/service/advancedcluster/model_advanced_cluster.go

```diff
@@ -933,3 +951,49 @@ func expandRegionConfigAutoScaling(tfList []any) *admin.AdvancedAutoScalingSetti
 	}
 	return &settings
 }
+
+func flattenAdvancedReplicationSpecsDS(ctx context.Context, apiRepSpecs []admin.ReplicationSpec20240710, d *schema.ResourceData, connV2 *admin.APIClient) ([]map[string]any, error) {
```

Comment:
I've split the flattener for replication spec for data source separately for now to keep the logic simple. Any improvements/refactoring would be done once update & relevant resource changes are implemented